chromium_src/ui/views/controls/button/md_text_button.cc

```diff
@@ -308,23 +295,25 @@ void MdTextButton::UpdateColors() {
   }
 }
 
-void MdTextButton::OnPaintBackground(gfx::Canvas* canvas) {
-  // Set brave-style hover colors
-  MdTextButtonBase::OnPaintBackground(canvas);
-  if (GetStyle() == ui::ButtonStyle::kProminent &&
-      (hover_animation().is_animating() || GetState() == STATE_HOVERED)) {
-    constexpr SkColor normal_color = gfx::kBraveColorBrand;
-    constexpr SkColor hover_color = SkColorSetA(normal_color, 0xFF * 0.2);
-    const SkAlpha alpha =
-        static_cast<SkAlpha>(hover_animation().CurrentValueBetween(0x00, 0xff));
-    const SkColor current_color =
-        color_utils::AlphaBlend(hover_color, normal_color, alpha);
-    cc::PaintFlags flags;
-    flags.setColor(current_color);
-    flags.setStyle(cc::PaintFlags::kFill_Style);
-    flags.setAntiAlias(true);
-    canvas->DrawRoundRect(gfx::RectF(GetLocalBounds()), GetCornerRadiusValue(),
-                          flags);
+void MdTextButton::SetStyle(ui::ButtonStyle button_style) {
+  MdTextButtonBase::SetStyle(button_style);
+
+  // Map to our style.
+  switch (GetStyle()) {
+    case ui::ButtonStyle::kProminent:
+      SetKind(kPrimary);
```

Comment:
I don't understand what we're doing here, why do we have `Kind` that maps directly to upstream styles? Why wouldn't we just use the upstream style names?

Reply:
cc: @simonhong

Reply:
Right, we could `ui::ButtonStyle` instead of `Kinde` now. then Patch file for `SetStyle` doesn't need anymore. Working on it.

Reply:
Removed `Kind` via  a951820f95fdade377b64fcc27947f494bb0682a